### ISSUE TITLE
Counting bug in Test::Harness with unexpected early exit

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8934]
Migrated from [rt.perl.org#43267](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=43267) (status was 'resolved')

Searchable as RT43267$

**Comment by p5pRT** [comment]
### From @doughera88

This is a duplicate of a problem I have already reported to rt\.cpan\.org\.
In case anyone wants to check\, it is filed there under

&nbsp;    http&#8203;://rt\.cpan\.org/Ticket/Display\.html?id=27377

I'm sending it to perlbug because Test&#8203;::Harness is dual\-lived\, and I hope 
this can be fixed before 5\.10 is released\.

Here's the original report&#8203;:

Test&#8203;::Harness gives wrong numbers when the test script dies unexpectedly\.
This happens for both bleadperl and perl\-5\.8\.8\.

Consider the following simple test file bug\.t&#8203;:

&nbsp;    use Test;
&nbsp;    BEGIN \{ plan tests => 6 \}
&nbsp;    ok\(1\);
&nbsp;    ok\(0\);
&nbsp;    ok\(0\);
&nbsp;    ok\(0\);
&nbsp;    die\("unexpected crash\!"\);
&nbsp;    ok\(5\);
&nbsp;    ok\(6\);

Running \./perl \-Ilib utils/prove bug\.t produces the incorrect result&#8203;:

&nbsp;    bug\.\.\.\.NOK 2/6\# Failed test 2 in bug\.t at line 4                             
&nbsp;    \#  bug\.t line 4 is&#8203;: ok\(0\);
&nbsp;    bug\.\.\.\.NOK 3/6\# Failed test 3 in bug\.t at line 5                             
&nbsp;    \#  bug\.t line 5 is&#8203;: ok\(0\);
&nbsp;    bug\.\.\.\.NOK 4/6\# Failed test 4 in bug\.t at line 6                             
&nbsp;    \#  bug\.t line 6 is&#8203;: ok\(0\);
&nbsp;    unexpected crash\! at bug\.t line 7\.
&nbsp;    bug\.\.\.\.dubious                                                               
&nbsp;        Test returned status 255 \(wstat 65280\, 0xff00\)
&nbsp;    DIED\. FAILED tests 2\-6
&nbsp;        Failed 5/6 tests\, 16\.67% okay
&nbsp;    Failed Test Stat Wstat Total Fail  List of Failed
&nbsp;    \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
&nbsp;    bug\.t        255 65280     6    7  2\-6
&nbsp;    Failed 1/1 test scripts\. 5/6 subtests failed\.
&nbsp;    Files=1\, Tests=6\,  0 wallclock secs \( 0\.01 cusr \+  0\.00 csys =  0\.01 CPU\)
&nbsp;    Failed 1/1 test programs\. 5/6 subtests failed\.

Note in the table that it claims there are 6 tests \(correct\) but that 7 failed\.
However\, in the summary\, it correctly notes that 5 out of the 6 tests
failed\.

perl\-5\.8\.8 fails in a similar way\, but with the added incorrect
percentage of 116\.67% failing\.

&nbsp;    Failed Test Stat Wstat Total Fail  Failed  List of Failed
&nbsp;    \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
&nbsp;    bug\.t        255 65280     6    7 116\.67%  2\-6

I speculate that it's double\-counting all tests after the die\(\)\.
For an extreme example\, consider the output I get from running 'make
test' for parrot&#8203;:

&nbsp;    Failed Test                       Stat Wstat Total Fail  Failed  List of Failed
&nbsp;    \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
&nbsp;    t/compilers/pge/p5regex/p5rx\.t       0     3   960 1466 152\.71%  228\-960
&nbsp;    t/compilers/pge/p6regex/01\-regex\.    0     3    ??   ??       %  ??
&nbsp;    t/examples/shootout\.t                1   256    20    1   5\.00%  16
&nbsp;    t/pmc/complex\.t                      5  1280    53    5   9\.43%  32 42\-43 46\-47
&nbsp;    t/src/hash\.t                         1   256    11    1   9\.09%  6
&nbsp;    t/src/io\.t                         255 65280    20   18  90\.00%  12\-20
&nbsp;     \(4 subtests UNEXPECTEDLY SUCCEEDED\)\, 11 tests and 277 subtests skipped\.
&nbsp;    Failed 6/304 test scripts\, 98\.03% okay\. 449/6669 subtests failed\, 93\.27% okay\.

where the numbers listed as faling for p5rx\.t and io\.t are exactly twice the number
you get from the "List of Failed"\.  Note too that there is no obvious
way to get the number '449' in the summary from any of the numbers in
the table above\!

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @smpeters

On Wed Jun 20 11&#8203;:32&#8203;:43 2007\, doughera wrote&#8203;:
> This is a duplicate of a problem I have already reported to
> rt\.cpan\.org\.
> In case anyone wants to check\, it is filed there under
> 
>     http&#8203;://rt\.cpan\.org/Ticket/Display\.html?id=27377
> 
> I'm sending it to perlbug because Test&#8203;::Harness is dual\-lived\, and I
> hope
> this can be fixed before 5\.10 is released\.
> 
> Here's the original report&#8203;:
> 
> Test&#8203;::Harness gives wrong numbers when the test script dies
> unexpectedly\.
> This happens for both bleadperl and perl\-5\.8\.8\.
> 
> Consider the following simple test file bug\.t&#8203;:
> 
>     use Test;
>     BEGIN \{ plan tests => 6 \}
>     ok\(1\);
>     ok\(0\);
>     ok\(0\);
>     ok\(0\);
>     die\("unexpected crash\!"\);
>     ok\(5\);
>     ok\(6\);
> 
> Running \./perl \-Ilib utils/prove bug\.t produces the incorrect result&#8203;:
> 
>     bug\.\.\.\.NOK 2/6\# Failed test 2 in bug\.t at line 4
>     \#  bug\.t line 4 is&#8203;: ok\(0\);
>     bug\.\.\.\.NOK 3/6\# Failed test 3 in bug\.t at line 5
>     \#  bug\.t line 5 is&#8203;: ok\(0\);
>     bug\.\.\.\.NOK 4/6\# Failed test 4 in bug\.t at line 6
>     \#  bug\.t line 6 is&#8203;: ok\(0\);
>     unexpected crash\! at bug\.t line 7\.
>     bug\.\.\.\.dubious
>         Test returned status 255 \(wstat 65280\, 0xff00\)
>     DIED\. FAILED tests 2\-6
>         Failed 5/6 tests\, 16\.67% okay
>     Failed Test Stat Wstat Total Fail  List of Failed
>    


---
>     bug\.t        255 65280     6    7  2\-6
>     Failed 1/1 test scripts\. 5/6 subtests failed\.
>     Files=1\, Tests=6\,  0 wallclock secs \( 0\.01 cusr \+  0\.00 csys =
> 0\.01 CPU\)
>     Failed 1/1 test programs\. 5/6 subtests failed\.
> 
> Note in the table that it claims there are 6 tests \(correct\) but that
> 7 failed\.
> However\, in the summary\, it correctly notes that 5 out of the 6 tests
> failed\.
> 
> perl\-5\.8\.8 fails in a similar way\, but with the added incorrect
> percentage of 116\.67% failing\.
> 
>     Failed Test Stat Wstat Total Fail  Failed  List of Failed
>    


---
>     bug\.t        255 65280     6    7 116\.67%  2\-6
> 
> I speculate that it's double\-counting all tests after the die\(\)\.
> For an extreme example\, consider the output I get from running 'make
> test' for parrot&#8203;:
> 
>     Failed Test                       Stat Wstat Total Fail  Failed
> List of Failed
>    


---
>     t/compilers/pge/p5regex/p5rx\.t       0     3   960 1466 152\.71%
> 228\-960
>     t/compilers/pge/p6regex/01\-regex\.    0     3    ??   ??       %
> ??
>     t/examples/shootout\.t                1   256    20    1   5\.00%
> 16
>     t/pmc/complex\.t                      5  1280    53    5   9\.43%
> 32 42\-43 46\-47
>     t/src/hash\.t                         1   256    11    1   9\.09%  6
>     t/src/io\.t                         255 65280    20   18  90\.00%
> 12\-20
>      \(4 subtests UNEXPECTEDLY SUCCEEDED\)\, 11 tests and 277 subtests
> skipped\.
>     Failed 6/304 test scripts\, 98\.03% okay\. 449/6669 subtests failed\,
> 93\.27% okay\.
> 
> where the numbers listed as faling for p5rx\.t and io\.t are exactly
> twice the number
> you get from the "List of Failed"\.  Note too that there is no obvious
> way to get the number '449' in the summary from any of the numbers in
> the table above\!
> 

Its listed as resolved in the Test\-Harness queue\, so we'll resolve it
here too\.

Steve

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 3 Mar 2008\, Steve Peters via RT wrote&#8203;:

> On Wed Jun 20 11&#8203;:32&#8203;:43 2007\, doughera wrote&#8203;:
> > This is a duplicate of a problem I have already reported to
> > rt\.cpan\.org\.
> > In case anyone wants to check\, it is filed there under
> > 
> >     http&#8203;://rt\.cpan\.org/Ticket/Display\.html?id=27377
> > 
> > I'm sending it to perlbug because Test&#8203;::Harness is dual\-lived\, and I
> > hope
> > this can be fixed before 5\.10 is released\.

> > I speculate that it's double\-counting all tests after the die\(\)\.

> Its listed as resolved in the Test\-Harness queue\, so we'll resolve it
> here too\.

Alas\, it's not resolved\.  Or\, more precisely\, it's resolved in the 
Test&#8203;::Harness 3\.x release\, but not in the 2\.64 release included with 
perl\-5\.10\.0\.

Once it became clear that Test&#8203;::Harness 3 wouldn't ship with perl\-5\.10\, I 
did take a look at fixing it myself\, but I confess I quickly got lost and 
gave up\.

If this is a serious issue for anyone\, they should probably consider 
upgrading to T&#8203;::H 3 instead of trying to fix this old code\.

I'll go in to RT and mark it "stalled"\, which probably captures the 
essence of the situation\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
@doughera88 - Status changed from 'resolved' to 'stalled'

**Comment by p5pRT** [comment]
### From andy@hexten.net

On 3 Mar 2008\, at 17&#8203;:09\, Andy Dougherty wrote&#8203;:
>> Its listed as resolved in the Test\-Harness queue\, so we'll resolve it
>> here too\.
>
> Alas\, it's not resolved\.  Or\, more precisely\, it's resolved in the
> Test&#8203;::Harness 3\.x release\, but not in the 2\.64 release included with
> perl\-5\.10\.0\.
>
> Once it became clear that Test&#8203;::Harness 3 wouldn't ship with  
> perl\-5\.10\, I
> did take a look at fixing it myself\, but I confess I quickly got  
> lost and
> gave up\.

I assumed that Test&#8203;::Harness 3\.xx would probably make it into 5\.10 at  
some point\. Is that problematic?

> If this is a serious issue for anyone\, they should probably consider
> upgrading to T&#8203;::H 3 instead of trying to fix this old code\.


Yes\, probably :\)

\-\- 
Andy Armstrong\, Hexten

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'stalled' to 'open'

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 3 Mar 2008\, Andy Armstrong via RT wrote&#8203;:

> > Alas\, it's not resolved\.  Or\, more precisely\, it's resolved in the 
> > Test&#8203;::Harness 3\.x release\, but not in the 2\.64 release included with 
> > perl\-5\.10\.0\.
>>
> > Once it became clear that Test&#8203;::Harness 3 wouldn't ship with  
> > perl\-5\.10\, I
> > did take a look at fixing it myself\, but I confess I quickly got  
> > lost and
> > gave up\.
> 
> I assumed that Test&#8203;::Harness 3\.xx would probably make it into 5\.10 at  
> some point\. Is that problematic?

Sorry\, I didn't mean to be misleading\.  I should have been more specific\.  
I meant to say "wouldn't ship with 5\.10\.0"\.

Whether it should ship with 5\.10\.1 \(or \.2\, etc\.\) is a different matter\.  
The default output is different enough that I can imagine it being an 
issue in some cases where someone might be relying on the old output 
format\.  However\, I don't have a sense of how widespread or important such 
breakage might be\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From andy@hexten.net

On 3 Mar 2008\, at 17&#8203;:57\, Andy Dougherty wrote&#8203;:
> Whether it should ship with 5\.10\.1 \(or \.2\, etc\.\) is a different  
> matter\.
> The default output is different enough that I can imagine it being an
> issue in some cases where someone might be relying on the old output
> format\.  However\, I don't have a sense of how widespread or  
> important such
> breakage might be\.


We did quite a lot of legwork during the development of it to attempt  
to identify such problems\. Of course\, there's no way of reaching all  
the places where it's used behind closed doors\, as it were\.

It's an issue that's going to come to a head at some point; I don't  
think anyone fancies going back and trying to maintain 2\.64 :\)

\-\- 
Andy Armstrong\, Hexten

**Comment by p5pRT** [comment]
### From @schwern

Andy Dougherty wrote&#8203;:
> Whether it should ship with 5\.10\.1 \(or \.2\, etc\.\) is a different matter\.  
> The default output is different enough that I can imagine it being an 
> issue in some cases where someone might be relying on the old output 
> format\.  However\, I don't have a sense of how widespread or important such 
> breakage might be\.

The summary display portion of TH 2 is some of the hairest bits of the code\, 
porting it over will be prohibitive\.  Anyone scraping it\.\.\. well\, they 
shouldn't be surprised if the formatting changes and their stuff breaks\.

TH 3 has programmatic facilities for getting results and the ability to write 
your own formatters that's far\, far nicer than screen scraping\.  Holding it 
back because of programs that are scraping results from TH 2 seems backwards\.

FWIW\, there's been no great outcry from the CPAN version since TH 3 has been 
released\.

BTW  I believe the original reasoning for keeping TH 3 out of 5\.10 was simply 
because TH 3 came out during the code freeze\.


\-\- 
E&#8203;: "Would you want to maintain a 5000 line Perl program?"
d&#8203;: "Why would you write a 5000 line program?"

**Comment by p5pRT** [comment]
### From andy@hexten.net

On 3 Mar 2008\, at 20&#8203;:10\, Michael G Schwern wrote&#8203;:
> BTW  I believe the original reasoning for keeping TH 3 out of 5\.10  
> was simply because TH 3 came out during the code freeze\.


That's my recollection too\.

\-\- 
Andy Armstrong\, Hexten

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 3 Mar 2008\, Michael G Schwern via RT wrote&#8203;:

> Andy Dougherty wrote&#8203;:
> > Whether it should ship with 5\.10\.1 \(or \.2\, etc\.\) is a different matter\.  
> > The default output is different enough that I can imagine it being an 
> > issue in some cases where someone might be relying on the old output 
> > format\.  However\, I don't have a sense of how widespread or important such 
> > breakage might be\.
> 
> The summary display portion of TH 2 is some of the hairest bits of the code\, 
> porting it over will be prohibitive\.  Anyone scraping it\.\.\. well\, they 
> shouldn't be surprised if the formatting changes and their stuff breaks\.
> 
> TH 3 has programmatic facilities for getting results and the ability to 
> write your own formatters that's far\, far nicer than screen scraping\.  
> Holding it back because of programs that are scraping results from TH 2 
> seems backwards\.

> BTW  I believe the original reasoning for keeping TH 3 out of 5\.10 was simply 
> because TH 3 came out during the code freeze\.

Right\.  That matches my recollection too\.  I didn't mean in any way to 
imply otherwise\.  I'm sorry some readers have apparently taken it that 
way\.  I was making no judgment of any sort\, merely stating that the 
version number shipped with 5\.10\.0 was 2\.64\.

However\, now that 5\.10\.0 has TH 2\, changing it to TH 3 in a maintenance 
release becomes one of those backwards\-compatibility\-vs\-forward\-progress 
tradeoffs that requires good balanced judgment\.

> FWIW\, there's been no great outcry from the CPAN version since TH 3 has 
> been released\.

That's good to hear\.  If that continues to be the case\, then upgrading in 
a maintenance release will probably be the right thing to do\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @jandubois

On Mon\, 03 Mar 2008\, Andy Armstrong wrote&#8203;:
> On 3 Mar 2008\, at 20&#8203;:10\, Michael G Schwern wrote&#8203;:
> > BTW  I believe the original reasoning for keeping TH 3 out of 5\.10
> > was simply because TH 3 came out during the code freeze\.
> 
> That's my recollection too\.

My augmented memory shows this message from "Andy Armstrong" on 11 Nov 2007
in thread "Re&#8203;: 5\.10 code freeze and dual\-life modules"&#8203;:

| On 12 Nov 2007\, at 16&#8203;:44\, jesse wrote&#8203;:
| > A major change\-the\-world release that happened after code freeze? As
| > much as I \_love\_ the new Test&#8203;::Harness \(prove \-j\)\, I sure hope it's 
| > not\.
|
| \+1 :\)
|
| I'd like the dust to settle around T&#8203;::H 3 a little before it sees that 
| kind of action\.

Cheers\,
\-Jan

**Comment by p5pRT** [comment]
### From @tux

On Mon\, 3 Mar 2008 20&#8203;:28&#8203;:19 \+0000\, Andy Armstrong \<andy@&#8203;hexten\.net> wrote&#8203;:

> On 3 Mar 2008\, at 20&#8203;:10\, Michael G Schwern wrote&#8203;:
> > BTW  I believe the original reasoning for keeping TH 3 out of 5\.10  
> > was simply because TH 3 came out during the code freeze\.
> 
> 
> That's my recollection too\.

That\, and that TH\-3 was not yet able to run perl5's CORE tests in parallel\,
which was one of the main reasons to get it into core\. Speed up our smokes\!

\-\- 
H\.Merijn Brand         Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using & porting perl 5\.6\.2\, 5\.8\.x\, 5\.10\.x  on HP\-UX 10\.20\, 11\.00\, 11\.11\,
& 11\.23\, SuSE 10\.1 & 10\.2\, AIX 5\.2\, and Cygwin\.       http&#8203;://qa\.perl\.org
http&#8203;://mirrors\.develooper\.com/hpux/            http&#8203;://www\.test\-smoke\.org
&nbsp;                        http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
### From andy@hexten.net

On 3 Mar 2008\, at 21&#8203;:22\, Jan Dubois wrote&#8203;:
> On Mon\, 03 Mar 2008\, Andy Armstrong wrote&#8203;:
>> On 3 Mar 2008\, at 20&#8203;:10\, Michael G Schwern wrote&#8203;:
>>> BTW  I believe the original reasoning for keeping TH 3 out of 5\.10
>>> was simply because TH 3 came out during the code freeze\.
>>
>> That's my recollection too\.
>
> My augmented memory shows this message from "Andy Armstrong" on 11  
> Nov 2007
> in thread "Re&#8203;: 5\.10 code freeze and dual\-life modules"&#8203;:
>
> | On 12 Nov 2007\, at 16&#8203;:44\, jesse wrote&#8203;:
> | > A major change\-the\-world release that happened after code  
> freeze? As
> | > much as I \_love\_ the new Test&#8203;::Harness \(prove \-j\)\, I sure hope  
> it's
> | > not\.
> |
> | \+1 :\)
> |
> | I'd like the dust to settle around T&#8203;::H 3 a little before it sees  
> that
> | kind of action\.


Most of which dust seems now to have settled :\)

\-\- 
Andy Armstrong\, Hexten

**Comment by p5pRT** [comment]
### From @rgs

On 03/03/2008\, Andy Dougherty \<doughera@&#8203;lafayette\.edu> wrote&#8203;:
> > BTW  I believe the original reasoning for keeping TH 3 out of 5\.10 was simply
>  > because TH 3 came out during the code freeze\.

And also because :
\* we wanted bugs to be shaked out TH3 before making it part of core
\* integrating large modules like this might take time and require
&nbsp;  adaptation and porting of numerous little things; not all tests
&nbsp;  in core are extremely well\-behaved\. \(but it turned out to be not
&nbsp;  that painful to integrate\)

>  However\, now that 5\.10\.0 has TH 2\, changing it to TH 3 in a maintenance
>  release becomes one of those backwards\-compatibility\-vs\-forward\-progress
>  tradeoffs that requires good balanced judgment\.
>
>
>  > FWIW\, there's been no great outcry from the CPAN version since TH 3 has
>  > been released\.
>
>
> That's good to hear\.  If that continues to be the case\, then upgrading in
>  a maintenance release will probably be the right thing to do\.

That's also my opinion\.

**Comment by p5pRT** [comment]
p5p@spam.wizbit.be - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @doughera88

On Wed\, 25 Jun 2008\, Bram via RT wrote&#8203;:

> According to our records\, your request regarding 
>   "Counting bug in Test&#8203;::Harness with unexpected early exit" 
> has been resolved\. 
> 
> If you have any further questions or concerns\, please respond to this message\.
> 
> For other topics\, please create a new ticket\.
> 
> Please don't feel obligated to say "Thanks" or "Kudos" or "I owe you a beer" \-\- if you respond to this message it will reopen the ticket\. If you must\, please send email directly to the person who handled your ticket\, and not to the tracking system\.
> 
> \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=43267 >

Is it really resolved?  I just tested it now on bleadperl
with the exact same script in the bug report\, and got a different
failure mode&#8203;:

&nbsp;    $ \./perl \-Ilib utils/prove bug\.t

&nbsp;    bug\.\.\.\.\.\.Can't locate Test\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;:
&nbsp;    /home/doughera/my/perl/lib/5\.11\.0/sun4\-solaris
&nbsp;    /home/doughera/my/perl/lib/5\.11\.0
&nbsp;    /home/doughera/my/perl/lib/site\_perl/5\.11\.0/sun4\-solaris
&nbsp;    /home/doughera/my/perl/lib/site\_perl/5\.11\.0 \.\) at bug\.t line 1\.
&nbsp;    BEGIN failed\-\-compilation aborted at bug\.t line 1\.
&nbsp;    bug\.\.\.\.\.\. Dubious\, test returned 2 \(wstat 512\, 0x200\)
&nbsp;     No subtests run 

&nbsp;    Test Summary Report
&nbsp;    \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
&nbsp;    bug\.t \(Wstat&#8203;: 512 Tests&#8203;: 0 Failed&#8203;: 0\)
&nbsp;      Non\-zero exit status&#8203;: 2
&nbsp;      Parse errors&#8203;: No plan found in TAP output
&nbsp;    Files=1\, Tests=0\,  0 wallclock secs \( 0\.01 usr \+  0\.00 sys =  0\.01 CPU\)
&nbsp;    Result&#8203;: FAIL

It seems my '\-Ilib' isn't getting passed through\.  @&#8203;INC only contains
the directories for my as\-yet\-uninstalled perl\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @doughera88

On Wed\, 25 Jun 2008\, Andy Dougherty wrote&#8203;:

> On Wed\, 25 Jun 2008\, Bram via RT wrote&#8203;:
> 
> > According to our records\, your request regarding 
> >   "Counting bug in Test&#8203;::Harness with unexpected early exit" 
> > has been resolved\. 

> Is it really resolved?  I just tested it now on bleadperl
> with the exact same script in the bug report\, and got a different
> failure mode&#8203;:
> 
>     $ \./perl \-Ilib utils/prove bug\.t
> 
>     bug\.\.\.\.\.\.Can't locate Test\.pm in @&#8203;INC \(@&#8203;INC contains&#8203;:

> It seems my '\-Ilib' isn't getting passed through\.  @&#8203;INC only contains
> the directories for my as\-yet\-uninstalled perl\.

Ah\.  This is a different issue \(see \[rt\.cpan\.org \#37124\]\)\.  The correct 
incantation is now&#8203;:

&nbsp;    \./perl \-Ilib utils/prove \-Ilib bug\.t

which results in the more reasonable&#8203;:

&nbsp;    bug\.\.\.\.\.\.\# Failed test 2 in bug\.t at line 4
&nbsp;    \#  bug\.t line 4 is&#8203;: ok\(0\);
&nbsp;    \# Failed test 3 in bug\.t at line 5
&nbsp;    \#  bug\.t line 5 is&#8203;: ok\(0\);
&nbsp;    \# Failed test 4 in bug\.t at line 6
&nbsp;    \#  bug\.t line 6 is&#8203;: ok\(0\);
&nbsp;    unexpected crash\! at bug\.t line 7\.
&nbsp;     Dubious\, test returned 255 \(wstat 65280\, 0xff00\)
&nbsp;     Failed 5/6 subtests 

&nbsp;    Test Summary Report
&nbsp;    \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
&nbsp;    bug\.t \(Wstat&#8203;: 65280 Tests&#8203;: 4 Failed&#8203;: 3\)
&nbsp;      Failed tests&#8203;:  2\-4
&nbsp;      Non\-zero exit status&#8203;: 255
&nbsp;      Parse errors&#8203;: Bad plan\.  You planned 6 tests but ran 4\.
&nbsp;    Files=1\, Tests=4\,  0 wallclock secs \( 0\.02 usr  0\.00 sys \+  0\.02 cusr  0\.00 csys =  0\.04 CPU\)
&nbsp;    Result&#8203;: FAIL

I'll go back in RT and make sure this bug stays closed\.

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu
